### DISCUSSION TITLE
Offline Blazor Webassembly

### DISCUSSION THREAD

**Discussion by BlagoCuljak** [MarimerLLC/csla#2418] [General]
Hello,
We have feature request, to make Blazor Webassembly app with offline support. 

There are tons of material of Internet, however, neither have answer to my situation.

Small CRM, has simple select combobox of existing contacts, and text field to input remarks.

We used Blazor and CSLA, can we somehow save database in local datastorage, fetch contacts from there, then save new contact inteaction, and when we get online, push new interaction to AppServer with database?

Has some had any experience with this type of arhitecture?

**Comment by rockfordlhotka** [comment]
The technology that makes Blazor wasm able to run offline is PWA - progressive web app. This is supported by Blazor, and is an open when creating a Blazor wasm project.

It has been a few months since I looked into doing this, and back then there were few (if any) client-side database engines that would run within a Blazor PWA. Hopefully that has changed over time. I'd check for SqlLite, as that's the most common client-side db engine for mobile. And there are others that are used in the Angular/React world that might now have Blazor support.
